uploadNeuroDB/NeuroDB/Log.pm

```diff
@@ -0,0 +1,192 @@
+    ############################################################
+    my $profile = "prod";
+    {
+     package Settings;
+     do "$ENV{LORIS_CONFIG}/.loris_mri/$profile";
+    }
+    $self->{'dbhr'} = $dbhr;
+    $self->{'origin'} = $origin;
+    $self->{'ProcessID'} = $processid;
+    return bless $self, $params;
+}
+
+
+################################################################
+################################################################
+## writeLog ####################################################
+## this is a useful function that will write Log messages ######
+## if the useTable is
+################################################################
+################################################################
+####Gets:
+=pod
+
+- Message, 
+- the type of message (error or log)
+- The origin of where the error is comging from
+- The processID/UploadID if it's coming from Imaging-uploader 
+--Todo: Name could be changed from processid to uploadid
+
+----
+ 
+if the write-to-log is enabled:
+- 1) It will write into the log file
+- 2) if the logtype is error then it will write into the error.log as well
+
+if the write-to-table is enabled:
+
+- 1) It will insert into the Log table:
+----a) the LogTypeID which is extracted using the origin
+----b) The processID/UploadID (which is the foreign key to 
+-----The mri-upload table--if it doesn't come from the mri-upload
+-----it will be empty..
+----c) CreateTime----The time that it is created...
+----d) The Log Message..
+----e) It will be true if it's an error message
+----f) And the CenterID
+
+
+=cut
+
+################################################################
+sub writeLog
+{
+
+    my $this           = shift;
+    my $use_log_table  = $Settings::use_log_table;
+    my $use_log_file   = $Settings::use_log_file;
+    my $is_error       = 0;
+    my ($message,$failStatus) = @_;
+ 
+    #############################################################
+    #######print out the message#################################
+    #############################################################
+    if ( $this->{ debug } )
+    {
+        print $message;
+    }
+
+    #############################################################
+    #######write the logs in the log file########################
+    #############################################################
+    if ( $use_log_file ) {
+        #############################################################
+    	################only if the file exists######################
+    	#############################################################
+    	if (($this->{logfile}) && (-e $this->{logfile})) {
+            $this->{LOG}->print($message);
+    	    if ($failStatus) {
+	            $this->{LOG}->print(
+	                "program exit status: $failStatus"
+	            );
+	            `cat $this->{logfile}  >> $this->{LogDir}/error.log`;
+	            `rm -f $this->{logfile} `;
```

Comment:
unix command should be written in a variable and executed using system($cmd).

my $rm_cmd = "cat $this->{logfile}  >> $this->{LogDir}/error.log"
system($rm_cmd)